### PR TITLE
Move builtins to `typegpu/data`

### LINKED ISSUE #630: [To be discussed] Move builtins to `typegpu/data`
Currently they are exported from `typegpu`, as in:

```ts
import tgpu, { builtin } from 'typegpu';
import * as d from 'typegpu/data';

const VertexOutput = d.struct({
  pos: builtin.position,
  uv: d.vec2f,
});
```

To me, given how they are currently meant to be used, it would make more sense to have them be in `typegpu/data`.

```ts
import * as d from 'typegpu/data';

const VertexOutput = d.struct({
  pos: d.builtin.position,
  uv: d.vec2f,
});
```

### PR DESCRIPTION
**Pull request by mhawryluk**
closes #630

### CODE REVIEW

File: packages/typegpu/src/data/index.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ export {
 } from './dataTypes';
 export { PUBLIC_sizeOf as sizeOf } from './sizeOf';
 export { PUBLIC_alignmentOf as alignmentOf } from './alignmentOf';
+export * from '../builtin';
```
Comment (iwoplaza) [inline packages/typegpu/src/data/index.ts]: Tiny nitpick, I would refrain from exporting * in a public module, as it is harder to control what is public and what is not. I know there are a lot of things exported from here, but could we list them here one-by-one anyway?

File: packages/typegpu/tests/builtin.test.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-import * as d from '../src/data';
 import {
   type BuiltinPosition,
   type BuiltinVertexIndex,
   type OmitBuiltins,
-  StrictNameRegistry,
   builtin,
-} from '../src/experimental';
+} from '../src/builtin';
+import * as d from '../src/data';
```
Comment (iwoplaza) [inline packages/typegpu/tests/builtin.test.ts:3]: Let's use the builtin inside `d`. That way, we can ensure that builtins are properly exported from a public entry point.

File: packages/typegpu/src/data/index.ts (hunk 1 above)
Comment (mhawryluk) [reply packages/typegpu/src/data/index.ts]: sure

File: packages/typegpu/src/data/index.ts [hunk 3]
```diff
@@ -107,6 +107,25 @@ export {
...
+  BuiltinSampleIndex,
+  BuiltinSampleMask,
+  BuiltinFragment,
+  BuiltinLocalInvocationId,
+  BuiltinLocalInvocationIndex,
+  BuiltinGlobalInvocationId,
+  BuiltinWorkgroupId,
+  BuiltinNumWorkgroups,
+  AnyBuiltin,
+  OmitBuiltins,
```
Comment (iwoplaza) [inline packages/typegpu/src/data/index.ts]: Seems more like an internal type, lets omit it for now. (haha, pun not intended)
```suggestion
```

**Review by iwoplaza — APPROVED** [review]
🏛️🏛️🏛️

**Pull request merged**